src/openms/include/OpenMS/ANALYSIS/OPENSWATH/OpenSwathOSWWriter.h

```diff
@@ -181,6 +191,22 @@ namespace OpenMS
             zErrMsg);
         sqlite3_free(zErrMsg);
       }
+
+      // Insert run_id information
+      std::stringstream sql_run;
+      sql_run << "INSERT INTO RUN (ID, FILENAME) VALUES ('"
+              << run_id_ << "', '"
+              << input_filename_ << "'); ";
+
+      // Execute SQL insert statement
+      rc = sqlite3_exec(db, sql_run.str().c_str(), callback, 0, &zErrMsg);
+      if( rc != SQLITE_OK )
+      {
+        throw Exception::IllegalArgument(__FILE__, __LINE__, OPENMS_PRETTY_FUNCTION,
+            zErrMsg);
+        sqlite3_free(zErrMsg);
```

Comment:
this free() is never executed --> memory leak
you'd have to rescue the content of zErrMsg to a local string (since it gets unwinded), then call free() and throw the exception using the local string